Manager/src/main/java/io/raspberrywallet/manager/modules/authorizationserver/AuthorizationServerModule.java

```diff
@@ -0,0 +1,137 @@
+    }
+    
+    @Override
+    public boolean check() {
+        if (hasInput("password")) {
+            try {
+                String password = getInput("password");
+                serverCredentials = new Credentials(walletUUID.toString(), password);
+                initialize();
+            } catch (RequestException e) {
+                return false;
+            }
+            return true;
+        }
+        else return serverCredentials != null;
+    }
+    
+    
+    private void initialize() throws RequestException {
+        if (!serverAPI.isRegistered(serverCredentials))
+            serverAPI.register(serverCredentials);
+        
+        if (!serverAPI.isLoggedIn())
+            serverAPI.login(serverCredentials);
+        
+        if (!serverAPI.secretIsSet(serverCredentials)) {
+            String randomSecret = getRandomString();
+            String encodedSecret = Base64.getEncoder().encodeToString(randomSecret.getBytes());
+            serverAPI.overwriteSecret(serverCredentials, encodedSecret);
+        }
+    }
+    
+    private String getRandomString() {
+        byte[] randomBytes = new byte[PASSWORD_SIZE_IN_BYTES];
+        random.nextBytes(randomBytes);
+        return new String(randomBytes);
+    }
+    
+    @Override
+    public byte[] encrypt(byte[] payload) throws EncryptionException {
+        try {
+            initialize();
+            
+            String encodedSecret = serverAPI.getSecret(serverCredentials);
+            Password password = new Password(Base64.getDecoder().decode(encodedSecret));
+            
+            AESEncryptedObject<ByteWrapper> encryptedSecret =
+                    CryptoObject.encrypt(new ByteWrapper(payload), password);
+            
+            return SerializationUtils.serialize(encryptedSecret);
+            
+        } catch (RequestException e) {
+            throw new EncryptionException(e.getMessage());
+        }
+    }
+    
+    @Override
+    public byte[] decrypt(byte[] keyPart) throws DecryptionException {
+        try {
+            String encodedSecret = serverAPI.getSecret(serverCredentials);
+            Password password = new Password(Base64.getDecoder().decode(encodedSecret));
+    
+            AESEncryptedObject<ByteWrapper> deserializedKeyPart =
+                    (AESEncryptedObject<ByteWrapper>) SerializationUtils.deserialize(keyPart);
+    
+            return CryptoObject.decrypt(deserializedKeyPart, password).getData();
+            
+        } catch (RequestException e) {
+            throw new DecryptionException(e.getMessage());
+        }
+    }
+    
+    /**
+     * There is no need to take actions before next encryption/decryption
+     * operations, so there is no point in implementing this method.
+     */
+    @Override
+    public void register() {}
+    
+    @Nullable
+    @Override
+    public String getHtmlUi() {
+        return "<input type=\"text\" name=\"password\">";
```

Comment:
so the login is the WalletUUID right ?

Reply:
yup, wallet manages it by itself, when there is no saved uuid for wallet, then a new one is generated and saved to file as plain text

Reply:
Looks good :)